front_end/panels/ai_chat/ui/settings/providers/OpenRouterSettings.ts

```
@@ -0,0 +1,537 @@
+  private fetchModelsStatus: HTMLElement | null = null;
+  private updateModelOptions: UpdateModelOptionsFunction;
+  private onSettingsSaved: () => void;
+  private onDialogHide: () => void;
+
+  // OAuth event handlers (stored for cleanup)
+  private handleOAuthSuccess: (() => void) | null = null;
+  private handleOAuthError: ((event: Event) => void) | null = null;
+  private handleOAuthLogout: (() => void) | null = null;
+
+  // Dynamically imported OAuth module
+  private OpenRouterOAuth: any = null;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)